### ISSUE TITLE
`error: linking uuid: FileNotFound` on bun install

### ISSUE THREAD

**Issue by ocheezyy-lw** [oven-sh/bun#7161]
### What version of Bun is running?

1.0.12+85c997513

### What platform is your computer?

Amazon Linux 2.4 x86 (Codebuild)

### What steps can reproduce the bug?

I was not able to reproduce this issue locally.

Create a bun project and install any of the following packages:
- json5
- uuid
- jsesc
- js-yaml
- regjsparser
- semver
- which

Run an AWS codebuild instance to build the project that executes: `bun install`

### What is the expected behavior?

Bun will install the packages without error

### What do you see instead?

```

1518 packages installed [51.38s]
--
152 |  
153 | error: linking json5: FileNotFound
154 |  
155 | error: linking jsesc: FileNotFound
156 |  
157 | error: linking js-yaml: FileNotFound
158 |  
159 | error: linking regjsparser: FileNotFound
160 |  
161 | error: linking semver: FileNotFound
162 |  
163 | error: linking resolve: FileNotFound
164 |  
165 | error: linking uuid: FileNotFound
166 |  
167 | error: linking json5: FileNotFound
168 |  
169 | error: linking which: FileNotFound
170 |  
171 | error: linking jsesc: FileNotFound
172 |  
173 | error: linking rollup: FileNotFound


```

### Additional information

I don't believe this issue is specific to codebuild itself but I don't know which environment could replicate it. I was also trying to rollback bun as I had no issues with it up until this point but somehow it's now failing with these errors and there seems to be no way to rollback or install a specific version.

**Comment by Jarred-Sumner** [comment]
@dylan-conway is this a regression?

**Comment by dylan-conway** [comment]
@ocheezyy-lw could we see your `package.json`? Interested in seeing all the dependencies

**Comment by ocheezyy-lw** [comment]
Of course!

```
  "dependencies": {
    "@apollo/client": "3.7.3",
    "@aws-sdk/client-cognito-identity-provider": "^3.266.1",
    "@fontsource/roboto": "^4.5.8",
    "@hookform/resolvers": "^2.9.10",
    "@mui/icons-material": "^5.10.16",
    "@mui/material": "^5.10.16",
    "@mui/styled-engine-sc": "^5.14.9",
    "@mui/styles": "^5.10.16",
    "@mui/x-data-grid": "^5.17.14",
    "@mui/x-data-grid-pro": "^5.17.14",
    "@mui/x-date-pickers": "^5.0.9",
    "@react-google-maps/api": "2.18.1",
    "@reduxjs/toolkit": "^1.9.1",
    "@vitejs/plugin-react-swc": "^3.3.2",
    "amazon-cognito-identity-js": "^6.1.2",
    "aws-appsync": "^4.1.9",
    "aws-appsync-auth-link": "^3.0.7",
    "aws-appsync-subscription-link": "^3.1.2",
    "date-fns": "^2.29.3",
    "docx": "^7.7.0",
    "file-saver": "^2.0.5",
    "file-system": "^1.2.2",
    "formik": "^2.2.9",
    "graphql": "^16.6.0",
    "libphonenumber-js": "^1.10.36",
    "lodash-es": "^4.17.21",
    "moment": "^2.29.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-dropzone": "^14.2.3",
    "react-hook-form": "^7.40.0",
    "react-icons": "^4.7.1",
    "react-papaparse": "^4.1.0",
    "react-redux": "^8.0.5",
    "react-router-dom": "^6.4.4",
    "react-tabs": "^6.0.1",
    "react-toastify": "^9.1.3",
    "recharts": "^2.4.3",
    "redux-persist": "^6.0.0",
    "styled-components": "^5.3.6",
    "ts-node": "^10.9.1",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@babel/preset-typescript": "^7.22.15",
    "@graphql-codegen/cli": "^2.16.5",
    "@graphql-codegen/client-preset": "^1.3.0",
    "@graphql-codegen/introspection": "^2.2.3",
    "@graphql-codegen/typescript-resolvers": "^2.7.13",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.4",
    "@types/lodash-es": "^4.17.9",
    "@types/node": "^20.9.1",
    "@types/react": "^18.2.21",
    "@types/react-dom": "^18.0.9",
    "@types/react-html-parser": "^2.0.2",
    "@types/react-redux": "^7.1.26",
    "@types/react-tabs": "^5.0.5",
    "@types/redux-saga": "^0.10.5",
    "@types/styled-components": "^5.1.26",
    "@typescript-eslint/eslint-plugin": "^5.45.1",
    "@typescript-eslint/parser": "^5.45.1",
    "bun-types": "^1.0.1",
    "cross-var": "^1.1.0",
    "dotenv-cli": "^7.0.0",
    "eslint": "^8.49.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-config-react-app": "^7.0.1",
    "eslint-config-standard-with-typescript": "^23.0.0",
    "eslint-loader": "^4.0.2",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-n": "^15.6.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-promise": "^6.1.1",
    "eslint-plugin-react": "^7.31.11",
    "html-react-parser": "^3.0.4",
    "husky": "^8.0.2",
    "jest": "^29.7.0",
    "jest-dom": "^4.0.0",
    "jest-environment-jsdom": "^29.7.0",
    "jest-transform-stub": "^2.0.0",
    "lint-staged": "^13.1.0",
    "prettier": "^2.8.0",
    "react-test-render": "^1.1.2",
    "rollup-plugin-visualizer": "^5.9.2",
    "ts-jest": "^29.1.1",
    "typescript": "^4.9.3",
    "vite": "^4.4.9",
    "vite-plugin-checker": "^0.6.2",
    "vite-plugin-compression": "^0.5.1",
    "vite-plugin-environment": "^1.1.3",
    "vite-plugin-eslint": "^1.8.1",
    "vite-plugin-svgr": "^3.2.0",
    "vite-tsconfig-paths": "^4.2.1",
    "web-vitals": "^3.1.0"
  },
  "optionalDependencies": {
    "fsevents": "^2.3.2"
  },
```

**Comment by squishyu** [comment]
Getting the same issue after updating to Bun v1.0.12
```
/app # bun install
bun install v1.0.12 (85c99751)
  📦 Installing [2592/3205] 
error: linking jest: FileNotFound
  📦 Installing [2615/3205] 
error: linking semver: FileNotFound
  📦 Installing [2763/3205] 
error: linking acorn: FileNotFound
  📦 Installing [2772/3205] 
error: linking json5: FileNotFound
  📦 Installing [3039/3205] 
error: linking jest-cli: FileNotFound
  📦 Installing [3098/3205] 
error: linking vite-node: FileNotFound
  📦 Installing [3240/3205] 
error: linking resolve: FileNotFound
  📦 Installing [3399/3205] 
error: linking vitest: FileNotFound
  📦 Installing [3501/3205] 
error: linking mime: FileNotFound
  📦 Installing [3600/3205] 
error: linking svgo: FileNotFound
  📦 Installing [3625/3205] 
error: linking acorn: FileNotFound
  📦 Installing [3721/3205] 
error: linking uuid: FileNotFound
  📦 Installing [3780/3205]
```

Installing with `--frozen-lockfile` then segfaults
```
/app # bun install --frozen-lockfile
bun install v1.0.12 (85c99751)
  📦 Installing [594/3205] Segmentation fault (core dumped)
```

Seems to be a regression, 1.0.11 worked just fine.

**Comment by patsaas** [comment]
This occurs since Bun 1.0.12 in the Cloudflare Pages build environment as well: Ubuntu 22.04.2 / x86_64.

**Comment by Jarred-Sumner** [comment]
Sorry about this.

Here's what we're doing:
1) We will release a fix for this tonight, in Bun v1.0.13.  
2) We will improve our test coverage for incremental installs with missing packages (i.e. some packages in node_modules are in there, but not all are). Most of our tests right now are for the empty state or the done state, but not the in-between state of some things are installed and some things are not.
